src/components/todo/TodoForm.tsx

```diff
@@ -1,17 +1,24 @@
 import React, { useState } from "react";
 import Input from "../shared/form/Input";
 import Button from "../shared/form/Button";
+import { Props } from "./TodoList";
 
-const TodoForm = () => {
+const TodoForm = ({ todos, setTodos }: Props) => {
   const [todotext, setTodoText] = useState<string>("");
-  const handelInputChange = (e: React.ChangeEvent<any>) => {
-    setTodoText(e.target.value);
+  // instead of using "e:any" always use "e:React.FormEvent"
+  const handleAdd = (e: React.FormEvent) => {
+    e.preventDefault();
+    if (todotext) {
+      setTodos([...todos, { id: Date.now(), todo: todotext, isDone: false }]);
```

Comment:
Keep the logic and state management separate.

ie 
``
const updatedTodos = [...todos];
const todo = { id: Date.now(), todo: todotext, isDone: false };
updatedTodos.push(todo);
setTodos(updatedTodos);
``